### PR TITLE
Membership approver groups

### PR DESCRIPTION
**Pull request by jacobsee**
Add approver group field to groups, allowing for the designation of group membership request approval to users that are not a member of the group itself. The use case is that a certain group may confer a high level of access to certain systems, which no users should have on a standing basis. This allows such a high-privileged group to remain 'normally empty', while still making it useful for break-glass access situations.

### CODE REVIEW

File: pkg/api/v1alpha1/auth.go [hunk 1]
```diff
@@ -294,6 +299,22 @@ func (r *Router) mwGroupAuthRequired(authRole mwAuthRole) gin.HandlerFunc {
 		}
 
 		queryMods := []qm.QueryMod{
+			models.GroupWhere.ID.EQ(id),
+		}
+
+		group, err := models.Groups(queryMods...).One(c.Request.Context(), r.DB)
```
Comment (fishnix) [inline pkg/api/v1alpha1/auth.go:305]: seems like `Find()` might be better here?

File: pkg/api/v1alpha1/auth.go [hunk 2]
```diff
@@ -413,6 +440,43 @@ func (r *Router) mwGroupAuthRequired(authRole mwAuthRole) gin.HandlerFunc {
...
+			}
+
+			adminGroups, err := models.Groups(qm.WhereIn("slug IN ?", ag...)).All(c.Request.Context(), r.DB)
+			if err != nil {
+				sendError(c, http.StatusInternalServerError, "error getting admin groups: "+err.Error())
+				return
+			}
+
+			for _, g := range adminGroups {
+				if contains(memberships, g.ID) {
```
Comment (fishnix) [inline pkg/api/v1alpha1/auth.go:463]: a map would perform better depending on the size of memberships

**Review by fishnix — COMMENTED** [review]
This looks good - I'm wondering if we should allow multiple approver groups?

**Comment by jacobsee** [comment]
> This looks good - I'm wondering if we should allow multiple approver groups?

I was thinking about that but landed on "we have group-of-groups functionality already and groups are cheap, so folks can create an _x-approvers_ group and configure that however they like". Otherwise we're adding a new table here and some more logic, particularly in the frontend. Thoughts?

File: pkg/api/v1alpha1/auth.go [hunk 3]
```diff
@@ -406,9 +402,9 @@ func (r *Router) mwGroupAuthRequired(authRole mwAuthRole) gin.HandlerFunc {
 		if authRole == AuthRoleAdminOrGroupAdmin {
 			isAdmin := false
 
-			memberships := make([]string, len(enumeratedMemberships))
-			for i, m := range enumeratedMemberships {
-				memberships[i] = m.GroupID
+			memberships := make(map[string]bool)
```
Comment (fishnix) [inline pkg/api/v1alpha1/auth.go:405]: nit picking but .... `map[string]struct{}` ?

**Review by fishnix — APPROVED** [review]

**Review by fishnix — APPROVED** [review]
lgtm!

**Pull request merged**